src/transform/src/lib.rs

```diff
@@ -251,11 +250,16 @@ impl Default for Optimizer {
                 ],
             }),
             Box::new(crate::reduction_pushdown::ReductionPushdown),
+            Box::new(crate::reduce_elision::ReduceElision),
+            Box::new(crate::projection_lifting::ProjectionLifting),
+            Box::new(crate::fusion::map::Map),
             Box::new(crate::cse::map::Map),
             Box::new(crate::projection_lifting::ProjectionLifting),
             Box::new(crate::join_implementation::JoinImplementation),
+            Box::new(crate::demand::Demand),
             Box::new(crate::fusion::project::Project),
             Box::new(crate::reduction::FoldConstants),
+            Box::new(crate::empty_map::EmptyMap),
```

Comment:
We should probably re-shake the order in which all of these optimizations happen, and in particular think through where we want / expect `ReductionPushdown`, but one thing I'd like to maintain is that `FoldConstants` stays the last optimization, as it is the one that determines whether the result is a `MirRelationExpr::Constant` or not. I get that `EmptyMap` doesn't change that, but the next edit to this sequence is more likely to append something after it.